codespell_lib/_codespell.py

```diff
@@ -1133,7 +1154,10 @@ def main(*args: str) -> int:
         )
         parser.print_help()
         return EX_USAGE
-    uri_ignore_words = parse_ignore_words_option(options.uri_ignore_words_list)
+    uri_ignore_words_lowercase, uri_ignore_words_cased = parse_ignore_words_option(
+        options.uri_ignore_words_list
+    )
+    uri_ignore_words = uri_ignore_words_lowercase | uri_ignore_words_cased
```

Comment:
Perhaps replace with:
```python
    uri_ignore_words = {itertools.chain(parse_ignore_words_option(options.uri_ignore_words_list))}
```

Reply:
this snippet didn't work for me as written but
```python
set().union(*parse_ignore_words_option(options.uri_ignore_words_list))
```
did work and has the advantage it doesn't need another import of `itertools`.

Reply:
Try this instead:
```python
    uri_ignore_words = set(itertools.chain(parse_ignore_words_option(options.uri_ignore_words_list)))
```
`itertools` is a standard library and chaining is what we do here.

```python
>>> import itertools
>>> set1 = {"a", "b", "c"}
>>> set2 = {"1", "2", "3"}
>>> {itertools.chain(set1, set2)}
{<itertools.chain object at 0x7fc52e24c700>}
>>> set(itertools.chain(set1, set2))
{'1', 'c', 'b', '3', 'a', '2'}
>>> 
```

Reply:
OK I can do that (but I think you need a `*` before `parse_ignore_words_option` for it to work).